pygen/pygen_src/isa/riscv_floating_point_instr.py

```diff
@@ -0,0 +1,150 @@
+"""
+Copyright 2020 Google LLC
+Copyright 2020 PerfectVIPs Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+"""
+
+import logging
+import vsc
+from pygen_src.isa.riscv_instr import riscv_instr
+from pygen_src.riscv_instr_pkg import (pkg_ins, riscv_fpr_t, riscv_instr_format_t,
+                                       riscv_instr_category_t)
+
+
+@vsc.randobj
+class riscv_floating_point_instr(riscv_instr):
+    def __init__(self):
+        super().__init__()
+        self.fs1 = vsc.rand_enum_t(riscv_fpr_t)
+        self.fs2 = vsc.rand_enum_t(riscv_fpr_t)
+        self.fs3 = vsc.rand_enum_t(riscv_fpr_t)
+        self.fd = vsc.rand_enum_t(riscv_fpr_t)
+        self.has_fs1 = vsc.bit_t(1)
+        self.has_fs2 = vsc.bit_t(1)
+        self.has_fs3 = vsc.bit_t(1)
+        self.has_fd = vsc.bit_t(1)
+        self.has_fs1 = 1
+        self.has_fs2 = 1
+        self.has_fs3 = 0
+        self.has_fd = 1
```

Comment:
looks like these are duplicated?

Reply:
Hi @udinator 
Actually, these lines are not duplicated. I have used it here to assign the default value. Since `vsc.bit_t(1)` has default value of 0.
Anyways, I have removed these statements and define these fields as `self.has_fs1 = vsc.bit_t(1, 1)` in order to retain the default values.

Reply:
ahh, that makes sense.
I think it might be more efficient to set the default value at the time of declaration, like you have updated this to.